### PR TITLE
Refactors assembly and chem_grenade code to /tg/ standard

### PR DESCRIPTION
**Pull request by DominikPanic**
Will need to be tested for balance.

:cl: LaKiller8
refactor: Assembly and chem grenade code has been refactored to /tg/ standard.
/:cl:

### CODE REVIEW

File: code/modules/assembly/voice.dm [hunk 1]
```diff
@@ -1,36 +1,109 @@
...
-	var/listening = 0
-	var/recorded	//the activation message
+	matter = list("metal" = 500, "glass" = 50)
+	attachable = TRUE
+	verb_say = "beeps"
+	verb_ask = "beeps"
+	verb_exclaim = "beeps"
+	var/listening = FALSE
+	var/recorded = "" //the activation message
+	var/mode = 1
```
Comment (Rohesie) [inline code/modules/assembly/voice.dm]:
```suggestion
	var/mode = INCLUSIVE_MODE
```

**Comment by Rohesie** [comment]
Merge conflict.

**Comment by Rohesie** [comment]
New conflict.

File: code/__HELPERS/mobs.dm [hunk 2]
```diff
@@ -153,4 +153,13 @@
 			. = capitalize(pick(GLOB.first_names_male)) + " " + capitalize(pick(GLOB.last_names))
 
 		if(!findname(.))
-			break
\ No newline at end of file
+			break
+
+
+/proc/get_mob_by_ckey(key)
```
Comment (Rohesie) [inline code/__HELPERS/mobs.dm]: Intended change?

File: code/modules/assembly/mousetrap.dm [hunk 3]
```diff
@@ -2,127 +2,106 @@
...
+	if(!armed)
+		return
+	var/datum/limb/affecting = null
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		switch(type)
+			if("feet")
+				if(!H.shoes)
+					affecting = H.get_limb(pick(BODY_ZONE_L_LEG, BODY_ZONE_R_LEG))
+					H.KnockDown(60)
```
Comment (Rohesie) [inline code/modules/assembly/mousetrap.dm]:
```suggestion
					H.KnockDown(3)
```
The /tg/ values are way higher than ours.

File: code/modules/assembly/mousetrap.dm [hunk 4]
```diff
@@ -2,127 +2,106 @@
...
+			if(BODY_ZONE_PRECISE_L_HAND, BODY_ZONE_PRECISE_R_HAND)
+				if(!H.gloves)
+					affecting = H.get_limb(type)
+					H.Stun(60)
```
Comment (Rohesie) [inline code/modules/assembly/mousetrap.dm]:
```suggestion
					H.Stun(2)
```

File: code/modules/assembly/proximity.dm [hunk 5]
```diff
@@ -2,159 +2,167 @@
 	name = "proximity sensor"
 	desc = "Used for scanning and alerting when someone enters a certain proximity."
 	icon_state = "prox"
-	matter = list("metal" = 800, "glass" = 200, "waste" = 50)
-	origin_tech = "magnets=1"
+	matter = list("metal" = 800, "metal" = 200)
```
Comment (Rohesie) [inline code/modules/assembly/proximity.dm]:
```suggestion
	matter = list("metal" = 800, "glass" = 200)
```
Or remove the second line.

File: code/modules/reagents/chem_splash.dm [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+	if(splash_holder.total_volume && affected_range >= 0)	//The possible reactions didnt use up all reagents, so we spread it around.
+		var/datum/effect_system/steam_spread/steam = new /datum/effect_system/steam_spread()
+		steam.set_up(10, 0, epicenter)
+		steam.attach(epicenter)
+		steam.start()
+
+		var/list/viewable = view(affected_range, epicenter)
+
+		var/list/accessible = list(epicenter)
+		for(var/i = 1; i <= affected_range; i++)
```
Comment (Rohesie) [inline code/modules/reagents/chem_splash.dm]:
```suggestion
		for(var/i = 1 to affected_range)
```

File: code/__HELPERS/mobs.dm (hunk 2 above)
Comment (DominikPanic) [reply code/__HELPERS/mobs.dm]: Yes it's used for the logging

**Pull request merged**